packages/camera/camera_avfoundation/ios/camera_avfoundation/Sources/camera_avfoundation/FLTCam.m

```diff
@@ -1313,9 +1314,45 @@ - (BOOL)setupWriterForPath:(NSString *)path {
   return YES;
 }
 
-- (void)setUpCaptureSessionForAudio {
+// this same function is also in video_player_avfoundation
+// configure application wide audio session manually to prevent overwriting
+// flag MixWithOthers by capture session, only change category if it is considered
+// as upgrade which means it can only enable ability to play in silent mode or
+// ability to record audio but never disables it, that could affect other plugins
+// which depend on this global state, only change category or options if there is
+// change to prevent unnecessary lags and silence
+// https://github.com/flutter/flutter/issues/131553
+static void upgradeAudioSessionCategory(AVAudioSessionCategory category,
+                                        AVAudioSessionCategoryOptions options,
+                                        AVAudioSessionCategoryOptions clearOptions) {
+  if (category == nil) {
```

Comment:
can you explain what `category` means here? is it the category that we want to upgrade to? or the category that we want to upgrade from?

Reply:
Yes, category to which it should upgrade.

Reply:
can you rename this to `requestedCategory`? Also when will we need to pass a `nil` category param? What does it mean by requesting a nil category? and why are we assigning `category = AVAudioSession.sharedInstance.category`  when it's nil